wgpu-core/src/hub.rs

```diff
@@ -121,32 +129,54 @@ impl<T, I: TypedId> ops::IndexMut<I> for Storage<T, I> {
 impl<T, I: TypedId> Storage<T, I> {
     pub fn contains(&self, id: I) -> bool {
         let (index, epoch, _) = id.unzip();
-        match self.map.get(index as usize) {
-            Some(&(_, storage_epoch)) => epoch == storage_epoch,
-            None => false,
+        match self.map[index as usize] {
+            Element::Occupied(_, storage_epoch) => epoch == storage_epoch,
+            Element::Vacant => false,
+            _ => unreachable!(),
         }
     }
 
-    pub fn insert(&mut self, id: I, value: T) -> Option<T> {
+    pub fn insert(&mut self, id: I, value: T) -> Element<T> {
```

Comment:
I suppose this is what made you have `pub` on the `Element`.
Let's just outright panic if there was something by that ID. I don't think we need to handle that gracefully anywhere.
Then we can return `T`